### PR TITLE
fix: disable limit orders state sync between tabs

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary
Context: https://cowservices.slack.com/archives/C0361CDG8GP/p1691675128738349

The bug was introduced in the recent jotai update: https://github.com/cowprotocol/cowswap/pull/2916

Before there was a solution to prevent sync jotai storages: https://github.com/pmndrs/jotai/pull/1004/files

In the `jotai@2.x` they removed this solution and now we have to patch storage to prevent the synchronization.

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/twap/state/twapOrdersListAtom.ts [hunk 1]
```diff
@@ -14,11 +16,7 @@ import { updateTwapOrdersList } from '../utils/updateTwapOrdersList'
 
 export type TwapOrdersList = { [key: string]: TwapOrderItem }
 
-export const twapOrdersAtom = atomWithStorage<TwapOrdersList>(
-  'twap-orders-list:v1',
-  {},
-  createJSONStorage(() => localStorage)
-)
+export const twapOrdersAtom = atomWithStorage<TwapOrdersList>('twap-orders-list:v1', {}, getJotaiIsolatedStorage())
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/twap/state/twapOrdersListAtom.ts:19]: I think the orders' list should be synced, no?

File: apps/cowswap-frontend/src/modules/twap/state/twapOrdersSettingsAtom.ts [hunk 2]
```diff
@@ -43,7 +44,7 @@ export const defaultTwapOrdersSettings: TwapOrdersSettingsState = {
 export const twapOrdersSettingsAtom = atomWithStorage<TwapOrdersSettingsState>(
   'twap-orders-settings-atom:v1',
   defaultTwapOrdersSettings,
-  createJSONStorage(() => localStorage)
+  getJotaiIsolatedStorage()
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/twap/state/twapOrdersSettingsAtom.ts:47]: Can you add that to limit orders settings as well?

**Review by elena-zh — COMMENTED** [review]
Changes LGTM!
However, I have a question related to the settings now on the Limit/Advanced orders pages.
Swap settings are isolated, but on the Limit/Advanced orders they are not. Should we leave it as it is or fix?

File: apps/cowswap-frontend/src/modules/twap/state/twapOrdersSettingsAtom.ts (hunk 2 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/twap/state/twapOrdersSettingsAtom.ts:47]: Sure

File: apps/cowswap-frontend/src/modules/twap/state/twapOrdersListAtom.ts (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/twap/state/twapOrdersListAtom.ts:19]: No, https://github.com/cowprotocol/cowswap/pull/2903

**Comment by shoom3301** [comment]
@elena-zh @alfetopito isolated limit and advanced settings as well

**Comment by shoom3301** [comment]
Reopened the PR into hotfix branch https://github.com/cowprotocol/cowswap/pull/3041

**Pull request closed without merging**